apollo-federation/src/sources/connect/validation/test_data/validation/all_fields_selected.graphql

```diff
@@ -0,0 +1,47 @@
+extend schema
+  @link(
+    url: "https://specs.apollo.dev/federation/v2.10"
+    import: ["@key", "@external", "@requires"]
+  )
+  @link(url: "https://specs.apollo.dev/connect/v0.1", import: ["@connect"])
+
+type Query {
+  ts: [T]
+    @connect(
+      http: { GET: "http://test/ts" }
+      selection: """
+      id
+      child {
+        id
+      }
+      """
+    )
+}
+
+type T @key(fields: "id") {
+  id: ID!
+  unselected: String!
```

Comment:
Nitpick: there may be some additional cases that would be good to add tests for. For example, if the selection aliases a field, or unwraps a child field, or wraps a field.

Reply:
i'm not too concerned about this because we're using the same visitor used in expansion code, but i added a few more cases